deeposlandia/datasets/aerial.py

```diff
@@ -0,0 +1,144 @@
+                new_in_path = os.path.join(output_dir, 'images', new_in_filename)
+                tile.save(new_in_path.replace(".tif", ".png"))
+                result_dicts.append({"raw_filename": image_filename,
+                                     "image_filename": new_in_path})
+
+        if labelling:
+            label_filename = image_filename.replace("images/", "gt/")
+            img_out = Image.open(label_filename) 
+            buffer_tiles = []
+            for x in range(0, raw_img_size, self.tile_size):
+                for y in range(0, raw_img_size, self.tile_size):
+                    tile = img_out.crop((x, y,
+                                         x + self.tile_size, y + self.tile_size))
+                    tile = utils.resize_image(tile, self.image_size)
+                    img_id = int((raw_img_size / self.tile_size
+                                  * x / self.tile_size
+                                  + y / self.tile_size))
+                    basename_decomp = os.path.splitext(
+                        os.path.basename(image_filename))
+                    new_out_filename = (basename_decomp[0] + '_' +
+                                       str(img_id) + basename_decomp[1])
+                    new_out_path = os.path.join(output_dir, 'labels',
+                                                new_out_filename)
+                    tile.save(new_out_path.replace(".tif", ".png"))
+                    labels = utils.label_building(tile,
+                                                  self.label_ids,
+                                                  dataset='aerial')
+                    result_dicts[img_id]["label_filename"] = new_out_path
+                    result_dicts[img_id]["labels"] = labels
+
+        return result_dicts
+
+    def populate(self, output_dir, input_dir, nb_images=None,
+                 aggregate=False, labelling=True):
+        """ Populate the dataset with images contained into `datadir` directory
+
+        Parameters
+        ----------
+        output_dir : str
+            Path of the directory where the preprocessed image must be saved
+        input_dir : str
+            Path of the directory that contains input images
+        nb_images : integer
+            Number of images to be considered in the dataset; if None, consider the whole
+        repository
+        aggregate : bool
+            Label aggregation parameter, useless for this dataset, but kept for
+        class method genericity
+        labelling: boolean
+            If True labels are recovered from dataset, otherwise dummy label are generated
+        """
+        image_list = os.listdir(os.path.join(input_dir, "images"))
+        image_list_longname = [os.path.join(input_dir, "images", l)
+                               for l in image_list if not l.startswith('.')][:nb_images]
+        utils.logger.info(("Getting {} images to preprocess..."
+                           "").format(len(image_list_longname)))
```

Comment:
The logger should use the "%s" format string, not the `format` method.

e.g. `logger.info("Save the file '%s'", filename)`

Reply:
Have you got a source, or a pep, that advises not to use the `format` method?

Reply:
And what about the f-string method? (see [PEP498](https://www.python.org/dev/peps/pep-0498/#how-to-denote-f-strings))